src/me/Pseudo/DecisionGraph/Main.java

```diff
@@ -1,80 +1,89 @@
 package me.Pseudo.DecisionGraph;
 
 import java.io.File;
-import java.util.ArrayList;
 import java.util.Scanner;
 
 public class Main {
 
 	public static void main(String[] args) {
-		// Execution arguments from cmd line
-		if(args.length == 2) {
+		
+		// traverse <file>
+		if(args.length == 2 || args.length == 3) {
 			
-			if(args[0].equalsIgnoreCase("traverse") || args[0].equalsIgnoreCase("tv")) {
-				
-				DecisionGraph dg = null;
-				try {
-					dg = new DecisionGraph(new File(args[1]));
-				}catch(Exception e) {
-					e.printStackTrace();
-				}
-				
-				traverse(dg);
-				
+			// Check arguments
+			if(!(args[0].equals("tv") || args[0].equals("traverse"))) {
+				System.out.println("Invalid Arguments!");
+				return;
 			}
 			
-		}
-		// TODO Add system for interactive construction/traversal
-	}
-	
-	// Traverse from console
-	private static void traverse(DecisionGraph dg) {
-		
-		// Setup for console traversal
-		Scanner sc = new Scanner(System.in);
-		dg.startTraversing();
-		
-		// Start traversing graph
-		while(!dg.isEndpoint()) {
-			
-			// Show text and available responses
-			System.out.println("Text: " + dg.text());
-			ArrayList<String> responses = dg.getResponses();
-			System.out.println("Responses: " + responses.toString());
+			// Attempt to load graph
+			DecisionGraph dg = null;
+			try {
+				dg = DecisionGraph.fromFile(new File(args[1]));
+			}catch(Exception e) {
+				e.printStackTrace();
+				return;
+			}
 			
-			// Request a choice
-			System.out.print("Choice: ");
-			String res = sc.nextLine();
+			// Assign entry point to be root or user-defined node
+			String entry = args.length == 3 ? args[2] : dg.getRoot();
 			
-			// Invalid response?
-			if(!responses.contains(res)) {
-				System.err.println("Invalid");
-				continue; // Skip assertion
+			// Attempt to init at start point
+			try {
+				traverse(dg, entry);
```

Comment:
`traverse` will throw an `Exception` for an invalid entry point to the `DecisionGraph` and the root entry point can be assumed to be valid.